lib/src/_shared/PickerPopper.tsx

```diff
@@ -0,0 +1,121 @@
+import Paper, { PaperProps } from '@material-ui/core/Paper';
+import Popper, { PopperProps } from '@material-ui/core/Popper';
+import TrapFocus, { TrapFocusProps } from '@material-ui/core/Unstable_TrapFocus';
+import { useForkRef } from '@material-ui/core/utils'
+import { makeStyles } from '@material-ui/core/styles';
+import { useGlobalKeyDown, keycode } from './hooks/useKeyDown';
+import { IS_TOUCH_DEVICE_MEDIA } from '../constants/dimensions';
+import { executeInTheNextEventLoopTick } from '../_helpers/utils';
+import { TransitionProps } from '@material-ui/core/transitions/transition';
+
+export interface ExportedPickerPopperProps {
+  /**
+   * Popper props passed down to material-ui [Popper](https://material-ui.com/api/popper/#popper-api) component.
+   */
+  PopperProps?: Partial<PopperProps>;
+  /**
+   * Custom component for [Transition](https://material-ui.com/components/transitions/#transitioncomponent-prop).
+   */
+  TransitionComponent?: React.ComponentType<TransitionProps>;
+}
+
+export interface PickerPopperProps extends ExportedPickerPopperProps, PaperProps {
+  role: 'tooltip' | 'dialog';
+  TrapFocusProps?: Partial<TrapFocusProps>;
+  anchorEl: PopperProps['anchorEl'];
+  open: PopperProps['open'];
+  onClose: () => void;
+}
+
+export const useStyles = makeStyles(
+  theme => ({
+    popper: {
+      zIndex: theme.zIndex.modal,
+    },
+    paper: {
+      transformOrigin: 'top center',
+      '&:focus': {
+        outline: 'auto',
+        [IS_TOUCH_DEVICE_MEDIA]: {
+          outline: 0,
+        },
+      },
+    },
+    topTransition: {
+      transformOrigin: 'bottom center',
+    },
+  }),
+  { name: 'MuiPickersPopper' }
+);
+
+export const PickerPopper: React.FC<PickerPopperProps> = ({
+  role,
+  PopperProps,
+  TrapFocusProps,
+  TransitionComponent = Grow,
+  open,
+  innerRef = null,
+  anchorEl,
+  children,
+  onClose,
+  ...other
+}) => {
+  const classes = useStyles();
+  const paperRef = React.useRef<HTMLElement>(null);
+  const handlePopperRef = useForkRef(paperRef, innerRef);
+
+  useGlobalKeyDown(open, {
+    [keycode.Esc]: onClose,
+  });
+
+  const handleBlur = () => {
+    // document.activeElement is updating on the next tick after `blur` called
+    executeInTheNextEventLoopTick(() => {
+      if (paperRef.current?.contains(document.activeElement)) {
+        return;
+      }
+
+      onClose();
+    });
+  };
+
+  return (
+    <Popper
+      transition
+      role={role}
+      open={open}
+      anchorEl={anchorEl}
+      placement="bottom"
+      className={clsx(classes.popper, PopperProps?.className)}
+      {...PopperProps}
+    >
+      {({ TransitionProps, placement }) => (
+        <TrapFocus
+          open={open}
+          disableAutoFocus={role === 'tooltip'}
+          disableEnforceFocus={role === 'tooltip'}
+          isEnabled={() => true}
+          getDoc={() => paperRef.current?.ownerDocument ?? document}
+        >
+          <TransitionComponent {...TransitionProps} timeout={350}>
```

Comment:
Where is 350 coming from, why overriding the default?

An order switch to allow customizations:

```suggestion
          <TransitionComponent timeout={350} {...TransitionProps}>
```

Reply:
Maybe it's a good opportunity to solve #1768? I don't know.